modules/carousel/jetpack-carousel.php

```diff
@@ -547,16 +682,28 @@ class_exists( 'Jetpack_AMP_Support' )
 					return $html;
 				}
 
+				$fake_root_tag = 'jetpack' . wp_rand( 10000, 99999 );
+				$charset       = get_option( 'blog_charset', 'utf-8' );
+
 				// Let's grab all containers from the HTML.
 				$dom_doc = new DOMDocument();
 
+				$dom_doc->encoding = $charset;
```

Comment:
I don't think this line is necessary. I added it to see what it might change, but I don't think it has any effect (the `<meta>` stuff below is sufficient). Testing may prove me wrong :)